### PR TITLE
Angular tree shaking

### PR DESCRIPTION
**Pull request by StephenCooper**
Changes required to resolve tree shaking issues when building with Angular.

Before these changes no code was tree shaken even if not used. Mainly because of the use of a module helper function that left all modules being included which then meant all the code was present.

Added an Angular module size project to keep tracking this going forward.

### CODE REVIEW

File: packages/ag-grid-angular/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "update-angular-components": "node updateGridAndColumnProperties.js && npm run prettier-fix"
   },
   "license": "MIT",
-  "module": "./dist/ag-grid-angular/fesm2022/ag-grid-community-angular.mjs",
+  "module": "./dist/ag-grid-angular/fesm2022/ag-grid-angular.mjs",
```
Comment (StephenCooper) [inline packages/ag-grid-angular/package.json:16]: Fix path to enable working with nx workspace

File: packages/ag-grid-angular/projects/ag-grid-angular/package.json [hunk 2]
```diff
@@ -3,11 +3,11 @@
...
   "license": "MIT",
   "peerDependencies": {
-    "ag-grid-community": "32.2.0",
-    "@angular/common": ">= 16.0.0",
-    "@angular/core": ">= 16.0.0"
+    "@angular/common": ">= 17.0.0",
+    "@angular/core": ">= 17.0.0"
   },
   "dependencies": {
+    "ag-grid-community": "32.2.0",
```
Comment (StephenCooper) [inline packages/ag-grid-angular/projects/ag-grid-angular/package.json:10]: Making ag-grid-community a proper dependency to be consistent with other frameworks.

File: packages/ag-grid-community/package.json [hunk 3]
```diff
@@ -8,8 +8,8 @@
   "exports": {
     ".": {
       "import": "./dist/package/main.esm.mjs",
-      "require": "./dist/package/main.cjs.js",
       "types": "./dist/types/src/main.d.ts",
+      "require": "./dist/package/main.cjs.js",
```
Comment (StephenCooper) [inline packages/ag-grid-community/package.json:12]: Prevents a warning from vite about the wrong order.

File: packages/ag-grid-community/src/validation/rules/colDefValidations.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { toStringWithNullUndefined } from '../logging';
 import type { Deprecations, OptionsValidator, Validations } from '../validationTypes';
 import { USER_COMP_MODULES } from './userCompValidations';
 
-const COLUMN_DEFINITION_DEPRECATIONS: Deprecations<ColDef | ColGroupDef> = {
+const COLUMN_DEFINITION_DEPRECATIONS: () => Deprecations<ColDef | ColGroupDef> = () => ({
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/validation/rules/colDefValidations.ts:8]: These changes enable these to be tree shaken when previously they were not. Just making them functions makes the difference here.

File: packages/ag-grid-enterprise/src/charts/chartComp/utils/chartParamsValidator.ts [hunk 5]
```diff
@@ -41,209 +41,191 @@ interface ValidationFunction<T, K extends keyof T = keyof T, V = T[K]> {
     warnIfFixed?: boolean;
 }
 
-export class ChartParamsValidator {
```
Comment (StephenCooper) [inline packages/ag-grid-enterprise/src/charts/chartComp/utils/chartParamsValidator.ts:44]: Converted from static class to standalone functions.

File: testing/module-size-angular/angular.json [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                    "configurations": {
+                        "production": {
+                            "budgets": [
+                                {
+                                    "type": "initial",
+                                    "maximumWarning": "250kB",
+                                    "maximumError": "500kB"
+                                },
+                                {
+                                    "type": "anyScript",
```
Comment (StephenCooper) [inline testing/module-size-angular/angular.json:36]: This value is set to match the current set of imported modules in the grid-wrapper.

File: packages/ag-grid-angular/projects/ag-grid-angular/ng-package.json [hunk 7]
```diff
@@ -4,5 +4,6 @@
     "lib": {
         "entryFile": "src/public-api.ts"
     },
-    "assets": ["./LICENSE.txt"]
+    "assets": ["./LICENSE.txt"],
+    "allowedNonPeerDependencies": ["ag-grid-community"]
```
Comment (seanlandsman) [inline packages/ag-grid-angular/projects/ag-grid-angular/ng-package.json:8]: Is this still necessary given this is now a full dependency?

File: packages/ag-grid-community/package.json (hunk 3 above)
Comment (seanlandsman) [reply packages/ag-grid-community/package.json:12]: vite or esbuild? (just curious)

**Review by seanlandsman — APPROVED** [review]
lgtm

File: packages/ag-grid-angular/projects/ag-grid-angular/ng-package.json (hunk 7 above)
Comment (StephenCooper) [reply packages/ag-grid-angular/projects/ag-grid-angular/ng-package.json:8]: That is required now to stop angular complaining that you have set a real dependency.

File: packages/ag-grid-community/package.json (hunk 3 above)
Comment (StephenCooper) [reply packages/ag-grid-community/package.json:12]: Yes probably esbuild actually from within the new Angular build setup.

File: testing/module-size-angular/src/app/app.component.html [hunk 8]
```diff
@@ -0,0 +1,336 @@
+<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
```
Comment (peterjrreynolds) [inline testing/module-size-angular/src/app/app.component.html]: Can delete this file?

Comment (StephenCooper) [reply testing/module-size-angular/src/app/app.component.html]: removed.

**Pull request merged**